### PR TITLE
support batch norm folding and fusing conv2d with bias add and activation

### PR DESCRIPTION
**Pull request by pyu10055**
Ported the batch norm folding logic from toco
Added support for utilizing grappler to fuse conv2d with bias add and activation function.

### CODE REVIEW

File: tfjs-converter/src/operations/executors/convolution_executor_test.ts [hunk 1]
```diff
@@ -202,14 +201,46 @@ describe('convolution', () => {
...
+          pad: 'same',
+          dataFormat: 'NHWC',
+          dilations: [2, 2],
+          bias: input3[0],
+          activation: 'relu',
+          preluActivationWeights: undefined
+        });
+      });
+    });
   });
```
Comment (annxingyuan) [inline tfjs-converter/src/operations/executors/convolution_executor_test.ts]: Should we also add a test for correctness of fused conv2d output?

File: tfjs-converter/python/tensorflowjs/converters/tf_saved_model_conversion_v2.py [hunk 2]
```diff
@@ -136,6 +161,24 @@ def optimize_graph(graph, output_node_names, output_graph, tf_version,
   optimized_graph = tf_optimizer.OptimizeGraph(
       config, meta_graph, cluster=get_cluster())
 
+  # batch norm folding
+  optimized_graph = fold_batch_norms(optimized_graph)
+
+  # set the device to CPU for all Conv2d nodes
+  for node in optimized_graph.node:
```
[the remark this answers is not in the document]
Comment (pyu10055) [reply tfjs-converter/python/tensorflowjs/converters/tf_saved_model_conversion_v2.py]: This is for enabling the grappler fusing processing, since it only do it for cpu op.

File: tfjs-converter/src/operations/executors/convolution_executor.ts [hunk 3]
```diff
@@ -23,150 +23,193 @@ import {InternalOpExecutor, Node} from '../types';
...
+          (getParamValue('fusedOps', node, tensorMap, context) as string[]);
+
+      const isBiasAdd = fusedOps[0] === 'biasadd';
+      const isPrelu = fusedOps[1] === 'prelu';
+      const isBatchNorm = fusedOps[0] === 'fusedbatchnorm';
+      const activationFunc = fusedOps[1];
+
+      const numArgs =
+          (getParamValue('numArgs', node, tensorMap, context) as number);
+      if (isBiasAdd && !isPrelu && numArgs !== 1) {
```
[the remark this answers is not in the document]
Comment (pyu10055) [reply tfjs-converter/src/operations/executors/convolution_executor.ts]: prelu freeze is not support yet, sine prelu is not a valid op, we need to figure out a different approach for it.

File: tfjs-converter/src/operations/executors/convolution_executor_test.ts (hunk 1 above)
Comment (pyu10055) [reply tfjs-converter/src/operations/executors/convolution_executor_test.ts]: yes make sense, will add.
